### PR TITLE
Port WindowsProfile and LinuxProfile structs from aks-engine into our code base.

### PR DESCRIPTION
**Pull request by yizhang4321**
This removes our dependency on those structs from aks-engine. Our end goal is to remove all aks-engine dependencies.

### CODE REVIEW

File: pkg/agent/datamodel/types.go [hunk 1]
```diff
@@ -18,6 +18,43 @@ import (
...
+	WindowsDockerVersion          string                `json:"windowsDockerVersion"`
+	Secrets                       []api.KeyVaultSecrets `json:"secrets,omitempty"`
+	SSHEnabled                    *bool                 `json:"sshEnabled,omitempty"`
+	EnableAutomaticUpdates        *bool                 `json:"enableAutomaticUpdates,omitempty"`
+	IsCredentialAutoGenerated     *bool                 `json:"isCredentialAutoGenerated,omitempty"`
+	EnableAHUB                    *bool                 `json:"enableAHUB,omitempty"`
+	WindowsPauseImageURL          string                `json:"windowsPauseImageURL"`
+	AlwaysPullWindowsPauseImage   *bool                 `json:"alwaysPullWindowsPauseImage,omitempty"`
+}
+
```
Comment (xuto2) [inline pkg/agent/datamodel/types.go:43]: FYI @AbelHu  For future windows cse migration to agentbaker.

Comment (AbelHu) [reply pkg/agent/datamodel/types.go:45]: Does this mean that AKS RP will start to use WindowsProfile under agentbaker?

Comment (yizhang4321) [reply pkg/agent/datamodel/types.go:45]: AKS RP and AgentBaker currently have their own data models. RP converts its data model to AgentBaker's before calling AgentBaker's functions.

**Pull request merged**